### PR TITLE
Update travis ci build process/scripts

### PR DESCRIPTION
**Pull request by mach6**
- Renamed run-travis-ci-tests.sh
- merged travis-config-for-javadoc-push.sh and
  push-javadoc-to-gh-pages.sh into
  travis-push-javadoc-to-gh-pages.sh
- updated logic for pushing javadocs to gh-pages
  such that it is not dependent on a separate
  git clone -- e.g. $HOME/SeLion. Now is uses
  the same git clone ($HOME/$repo/SeLion) that
  the build is done on and instead sets up a
  remote tracking branch against a
  'writable' remote
- bump maven javadoc plugin version
- Add logic in sonatype deploy script to check
  current active git branch is 'develop'

### CODE REVIEW

**Comment by mach6** [comment]
@sebady @sundaramrajendran please help me by reviewing this PR

**Comment by mach6** [comment]
@ysecaran fyi

File: utilities/travis-push-javadoc-to-gh-pages.sh [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    rm -rf html/java-docs/develop/server-apis/*
+    rm -rf html/java-docs/develop/client-apis/*
+    rm -rf html/java-docs/develop/codegen-apis/*
+    rm -rf html/java-docs/develop/dataproviders-apis/*
+
+    # copy the javadoc files to respective folders from develop branch into gh-pages branch
+    cp -R client/target/apidocs/* html/java-docs/develop/client-apis/
+    cp -R codegen/target/apidocs/* html/java-docs/develop/codegen-apis/
+    cp -R server/target/apidocs/* html/java-docs/develop/server-apis/
+    cp -R dataproviders/target/apidocs/* html/java-docs/develop/dataproviders-apis/
```
Comment (sebady) [inline utilities/travis-push-javadoc-to-gh-pages.sh:75]: Shouldn't the  "common" module be in this list?

Comment (mach6) [inline utilities/travis-push-javadoc-to-gh-pages.sh:75]: yes. there is an oversight here that dates back to when we created the common jar. I will fix it

**Comment by mach6** [comment]
this is merged

**Pull request closed without merging**